lib/mail/fields/content_disposition_field.rb

```diff
@@ -9,6 +10,7 @@ class ContentDispositionField < StructuredField
 
     def initialize(value = nil, charset = 'utf-8')
       self.charset = charset
+      value = value.dup if value.is_a?(String) && value.frozen?
       ensure_filename_quoted(value)
       super(CAPITALIZED_FIELD, strip_field(FIELD_NAME, value), charset)
       self.parse
```

Comment:
Perhaps `#ensure_filename_quoted` shouldn't be modifying `value` in-place, instead?